src/wallet/wallet.h

```diff
@@ -808,6 +809,9 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
 
     std::set<COutPoint> setLockedCoins GUARDED_BY(cs_wallet);
 
+    /** Validation event callback handler. */
+    std::unique_ptr<interfaces::Handler> m_handler;
```

Comment:
I don't really like this name `m_handler`, (or the class name `HandlerImpl`), since the name handler is generic for any interface handler. It's more verbose, but I think `m_validation_interface_handler` and `ValidationInterfaceHandlerImpl` are more explicit.